src/shared/bpf-program.c

```diff
@@ -28,6 +41,38 @@ int bpf_program_new(uint32_t prog_type, BPFProgram **ret) {
         return 0;
 }
 
+int bpf_program_new_from_bpffs_path(const char *path, enum bpf_attach_type attach_type, BPFProgram **ret) {
+        _cleanup_(bpf_program_unrefp) BPFProgram *p = NULL;
+        struct bpf_prog_info info = {};
```

Comment:
hmm, this is used as output param for bpf_program_get_info_by_fd(), i.e. is there any need to initialize this here? i'd assume that that function would fill in its data but not care about what is already in it

Reply:
@poettering 

It has to be initialized because `bpf_prog_info` stores input too.`BPF_OBJ_GET_INFO_BY_FD` syscall execution depends on which fields are already set.
E.g. to get map fds of a loaded program, a user must specify the number of ids in `nr_map_ids`  and a buffer `map_ids` with enough size to store  `nr_map_ids`:
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/kernel/bpf/syscall.c?h=v5.11-rc4#n3416

Reply:
can you add a brief comment about that to the declartion, so that this is clear when reading the sources?

Reply:
Added before function definition.